plugins/redis/redis.cpp

```diff
@@ -354,7 +354,7 @@ void * Connection::redisCommand(const char * format, ...)
 }
 int Connection::setTimeout(unsigned _timeout)
 {
-    struct timeval to = { _timeout/1000, (_timeout%1000)*1000 };
+    struct timeval to = { (int) _timeout/1000, (int) (_timeout%1000)*1000 };
```

Comment:
I think this should really be

{ (time_t)(_timeout/1000), (suseconds_t)((_timeout%1000)*1000) };

something similar above (and don't change the type of timeStillLeft)